components/scream/cime_config/namelist_defaults_scream.xml

```diff
@@ -0,0 +1,130 @@
+<?xml version="1.0"?>
+
+<namelist_defaults>
+
+  <file name="scream_input.yaml" format="yaml">
+    <Atmosphere Driver>
+      <Atmosphere Processes>
+        <Number of Entries>6</Number of Entries>
+        <Schedule Type>Sequential</Schedule Type>
+        <Process 0>
+          <Process Name>Homme</Process Name>
+          <Enable Output Fields Checks>False</Enable Output Fields Checks>
+          <Enable Input Fields Checks>False</Enable Input Fields Checks>
+          <Grid>Dynamics</Grid>
+        <\Process 0>
+        <Process 1>
+          <Process Name>SHOC</Process Name>
+          <Enable Output Fields Checks>False</Enable Output Fields Checks>
+          <Enable Input Fields Checks>False</Enable Input Fields Checks>
+          <Grid>Physics GLL</Grid>
+        <\Process 1>
+        <Process 2>
+          <Process Name>CldFraction</Process Name>
+          <Enable Output Fields Checks>False</Enable Output Fields Checks>
+          <Enable Input Fields Checks>False</Enable Input Fields Checks>
+          <Grid>Physics GLL</Grid>
+        <\Process 2>
+        <Process 3>
```

Comment:
I think this format has the disadvantage that, in order to xmlchange/yamlchange/paramchange an option for a particular process, you need to know where that process is in the DAG. E.g., to change "SPA Remap File", you would have to know that SPA is in the sublist "Process 3".

I think we might benefit from some changes in our yaml structure, so that you don't need to know this kind of detail. E.g., for the param above, one should be able to do just "Atmosphere Driver"->"Processes"->"SPA".

Reply:
I think it's reasonable for people doing xmlchanges to know what process they're trying to change - if they didn't, they don't have any business changing things! That said, I don't think we can expect them to know that SPA is "Process 3" - is there any way to list the processes by their *name* and to have process *number* be a property of each entry?

Reply:
Yes, I've been thinking about this for a while, but put it on the (way) back burner. See #843 . That, in addition to #1449 lays a reasonable path for how to make the declaration of atm procs specs independent of their organization/order.

Reply:
Good thoughts, and I did not address the user-mods aspect of things in this PR. If we like the general look of this prototype, we would not be using xmlchange since that only supports CIME case settings. It would likely be a (hopefully simple) homegrown script. Since it is homegrown, we would be free to add smarts to it so that names to things would not have to be fully qualified, ex:

`eamxx-config-change SPA::SPA_Remap_File $mynewfile`